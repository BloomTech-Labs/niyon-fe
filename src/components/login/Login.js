import React from "react";
import { useForm } from "react-hook-form";
import "./styles.scss";

const Login = () => {
  const { register, handleSubmit, errors } = useForm();
  const onSubmit = (data) => console.log(data);
  console.log(errors);
  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <div>
        <input
          type="email"
          placeholder="Email"
          name="email"
          ref={register({ required: true })}
        />
        {errors.email && (
          <p style={{ color: "orange", marginTop: 10 }}>"Email is required"</p>
        )}
      </div>
      <div>
        <input
          type="password"
          placeholder="Password"
          name="password"
          ref={register({ required: true })}
        />
        {errors.password && (
          <p style={{ color: "orange", marginTop: 10 }}>
            "Password is required"
          </p>
        )}
      </div>
      <button type="submit">Login</button>
    </form>
  );
};

export default Login;
