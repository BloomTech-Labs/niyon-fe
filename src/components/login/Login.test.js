import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Login from './Login';
import { act } from 'react-dom/test-utils';


describe('<Login /> component testing', () => {
  let component;
  let onSubmit;
  beforeEach(() => {
    onSubmit = jest.fn();
    component = shallow(<Login onSubmit={onSubmit} />);
  });

  afterEach(() => {
     jest.clearAllMocks();
  })

it('should pass snaps shot testing', () => {
      expect(toJSON(component)).toMatchSnapshot();
    });

 it('should render <Login /> component correctly', () => {
      console.log(component.debug());
      expect(component.exists()).toBe(true);     
    });

//  it('should render form correctly', () => {
//       expect(component.find('form').exists()).toBe(true);
//      });

//  it('should render the div container with class name "formWrap"', () => {
//       expect(component.find('div').at(0).exists()).toBe(true);
//       expect(component.find('div').at(0).hasClass('formWrap')).toBe(true);
//      });

//  it('should render the div container with class name "formLogin"', () => {
//       expect(component.find('div').at(1).exists()).toBe(true);
//       expect(component.find('div').at(1).hasClass('formLogin')).toBe(true);
//     });
//   it('should render the input for email correctly', () => {
//       expect(component.find('input[type="email"]').exists()).toBe(true);
//     });

//   it('should render the input for email correctly', () => {
//       expect(component.find('input[type="password"]').exists()).toBe(true);
//     });

//   it('should render submit button correctly with its text', () => {
//       const button = component.find('button');
//       expect(button.exists()).toBe(true);
//       expect(button.text()).toEqual('Login');
//     });

//     describe('Form submission', () => {
//         it('should submit when data filled', async () => {
//             const emailInput = component.find('input[type="email"]');
//             // const passwordInput = component.find('input[type="password"]');            
//             // emailInput.value = "test@email.com";
//             // emailInput.dispatchEvent(new Event("input"));
//           await act( async() => {
//               emailInput.simulate('change', { target: { name:'email', value: 'test@email.com'}});
//               component.find('button').simulate('submit');
           
//             });
//           // await act( async() => {            
//           //   component.find('button').simulate('submit');
//           //   expect(onSubmit).toHaveBeenCalled();
//           // })
//         });
    // });


});  