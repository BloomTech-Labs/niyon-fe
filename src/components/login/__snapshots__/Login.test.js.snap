// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> component testing should pass snaps shot testing 1`] = `
<form
  data-test="form"
  onSubmit={[Function]}
>
  <div
    className="formWrap"
  >
    <Header />
    <div
      className="formLogin"
    >
      <input
        data-test="email"
        name="email"
        placeholder="Email"
        type="email"
      />
      <input
        data-test="password"
        name="password"
        placeholder="Password"
        type="password"
      />
      <button
        data-test="submit"
        type="submit"
      >
        Login
      </button>
    </div>
  </div>
</form>
`;
